diff -durN -x autom4te.cache -x configure -x '*.m4' -x '*.orig' -x '*.rej' -x '*.user' orig-src/gcc-10.2.0/configure.ac toolchain_build/src/gcc-10.2.0/configure.ac
--- orig-gcc-10.2.0/configure.ac    2020-09-24 23:03:44.963540752 -0400
+++ gcc-10.2.0/configure.ac 2020-10-08 04:19:24.144928982 -0400
@@ -331,7 +331,7 @@
       *-*-elf* | *-*-sysv4* | *-*-unixware* | *-*-eabi* | hppa*64*-*-hpux* \
       | *-*-linux* | *-*-gnu* | frv-*-uclinux* | *-*-irix5* | *-*-irix6* \
       | *-*-netbsd* | *-*-openbsd* | *-*-freebsd* | *-*-dragonfly* \
-      | *-*-solaris2* | *-*-nto* | *-*-nacl*)
+      | *-*-solaris2* | *-*-nto* | *-*-nacl* | *-*-dgos*)
         case "${target}" in
           *-*-linux*aout* | *-*-linux*oldld*)
             ;;
diff -durN -x Makefile.in -x autom4te.cache -x configure -x crossconfig.m4 orig-src/gcc-10.2.0/gcc/configure.ac toolchain_build/src/gcc-10.2.0/gcc/configure.ac
--- orig-gcc-10.2.0/gcc/configure.ac        2020-09-24 23:03:44.811541784 -0400
+++ gcc-10.2.0/gcc/configure.ac     2020-10-08 03:58:06.822915078 -0400
@@ -2121,6 +2121,8 @@
 	if test -f ${srcdir}/config/$f
 	then
 		tmake_file_="${tmake_file_} \$(srcdir)/config/$f"
+	else
+		echo "*** Ignored tmake_file named '$f' that was expected to be at ${srcdir}/config/$f"
 	fi
 done
 tmake_file="${tmake_file_}${omp_device_property_tmake_file}"
diff -durN gcc-6.2.0/gcc/config/i386/t-x86_64-elf gcc-6.2.0.patched/gcc/config/i386/t-x86_64-dgos
--- gcc-6.2.0/gcc/config/i386/t-x86_64-dgos	1969-12-31 16:00:00.000000000 -0800
+++ gcc-6.2.0.patched/gcc/config/i386/t-x86_64-dgos	2016-09-30 11:55:53.602700086 -0700
@@ -0,0 +1,45 @@
+MULTILIB_OPTIONS  :=
+MULTILIB_DIRNAMES :=
+
+MULTILIB_OPTIONS  += m32
+MULTILIB_DIRNAMES += 32
+
+MULTILIB_OPTIONS  += mrtd
+MULTILIB_DIRNAMES += rtd
+
+# 32 bit code using up to 3 register parameters
+MULTILIB_OPTIONS  += mregparm=3
+MULTILIB_DIRNAMES += regparm3
+
+MULTILIB_OPTIONS  += mno-red-zone
+MULTILIB_DIRNAMES += no-red-zone
+
+MULTILIB_OPTIONS  += Wl,--no-dynamic-linker
+MULTILIB_DIRNAMES += no-dynamic-linker
+
+MULTILIB_OPTIONS  += mcmodel=medium/mcmodel=large
+MULTILIB_DIRNAMES += medium large
+
+# Required libraries in decreasing order of strangeness
+
+# 64 bit UEFI code
+MULTILIB_REQUIRED := mno-red-zone/mcmodel=large
+
+# 32 bit bootloader code
+MULTILIB_REQUIRED += m32/mrtd/mregparm=3
+
+# medium model 64 bit user code
+MULTILIB_REQUIRED += mcmodel=medium
+
+# 32 bit user code
+MULTILIB_REQUIRED += m32
+
+# Normal kernel code
+MULTILIB_REQUIRED += mno-red-zone
+
+MULTILIB_EXTRA_OPTS := fno-plt
+
+CRTSTUFF_T_CFLAGS := -fno-plt
+
+$(info doug16k wtf $(MULTILIB_OPTIONS))
+
diff -durN gcc-8.2.0-orig/libgcc/unwind-dw2.c gcc-8.2.0/libgcc/unwind-dw2.c
--- gcc-8.2.0-orig/libgcc/unwind-dw2.c	2018-08-14 11:21:32.915943395 -0400
+++ gcc-8.2.0/libgcc/unwind-dw2.c	2018-08-14 11:31:35.134075621 -0400
@@ -22,6 +22,8 @@
    see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
    <http://www.gnu.org/licenses/>.  */
 
+#pragma GCC target ("no-sse")
+
 #include "tconfig.h"
 #include "tsystem.h"
 #include "coretypes.h"
diff -durN -x Makefile.in -x autom4te.cache -x configure -x crossconfig.m4 orig-src/gcc-10.2.0/config.sub src/gcc-10.2.0/config.sub
--- gcc-10.2.0-orig/config.sub	2020-09-24 23:03:44.963540752 -0400
+++ gcc-10.2.0/config.sub	2020-09-24 23:47:48.567160367 -0400
@@ -1396,6 +1396,9 @@
 	linux*)
 		os=`echo $os |["gcc_2"]=" sed -e 's|linux|linux-gnu|'`
 		;;
+	dgos*)
+		os=dgos
+		;;
 	lynx*178)
 		os=lynxos178
 		;;
diff -durN -x Makefile.in -x autom4te.cache -x configure -x crossconfig.m4 orig-src/gcc-10.2.0/fixincludes/mkfixinc.sh src/gcc-10.2.0/fixincludes/mkfixinc.sh
--- gcc-10.2.0-orig/fixincludes/mkfixinc.sh	2020-09-24 23:03:44.967540724 -0400
+++ gcc-10.2.0/fixincludes/mkfixinc.sh	2020-09-25 04:12:01.660663077 -0400
@@ -11,6 +11,7 @@
 
 # Check for special fix rules for particular targets
 case $machine in
+    *-dgos* | \
     i?86-*-cygwin* | \
     i?86-*-mingw32* | \
     x86_64-*-mingw32* | \
diff -durN -x Makefile.in -x autom4te.cache -x configure -x crossconfig.m4 orig-src/gcc-10.2.0/gcc/config/dgos.h src/gcc-10.2.0/gcc/config/dgos.h
--- gcc-10.2.0-orig/gcc/config/dgos.h	1969-12-31 19:00:00.000000000 -0500
+++ gcc-10.2.0/gcc/config/dgos.h	2020-09-24 23:59:35.255646822 -0400
@@ -0,0 +1,28 @@
+/* Useful if you wish to make target-specific GCC changes. */
+#undef TARGET_DGOS
+#define TARGET_DGOS 1
+
+/* Default arguments when running
+   i686-dgos-gcc/x86_64-dgos-gcc toolchain */
+#undef LIB_SPEC
+#define LIB_SPEC "-lc" /* link against C standard library */
+
+/* Files that are linked before user code.
+   The %s tells GCC to look for these files in the library directory. */
+#undef STARTFILE_SPEC
+#define STARTFILE_SPEC "crt0.o%s crtbegin.o%s"
+
+/* Files that are linked after user code. */
+#undef ENDFILE_SPEC
+#define ENDFILE_SPEC "crtend.o%s"
+
+/* Additional predefined macros. */
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()      \
+  do {                                \
+    builtin_define ("__dgos__");      \
+    builtin_define ("__unix__");      \
+    builtin_assert ("system=dgos");   \
+    builtin_assert ("system=unix");   \
+    builtin_assert ("system=posix");   \
+  } while(0);
diff -durN -x Makefile.in -x autom4te.cache -x configure -x crossconfig.m4 orig-src/gcc-10.2.0/gcc/config.gcc src/gcc-10.2.0/gcc/config.gcc
--- gcc-10.2.0-orig/gcc/config.gcc	2020-09-24 23:03:44.827541676 -0400
+++ gcc-10.2.0/gcc/config.gcc	2020-09-24 23:57:46.495586995 -0400
@@ -675,6 +675,12 @@
 
 # Common parts for widely ported systems.
 case ${target} in
+*-*-dgos*)
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  use_gcc_stdint=wrap
+  ;;
 *-*-darwin*)
   tmake_file="t-darwin "
   tm_file="${tm_file} darwin.h"
@@ -2010,6 +2016,13 @@
 	done
 	TM_MULTILIB_CONFIG=`echo $TM_MULTILIB_CONFIG | sed 's/^,//'`
 	;;
+i[34567]86-*-dgos*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h newlib-stdint.h i386/i386elf.h dgos.h"
+	;;
+x86_64-*-dgos*)
+	tmake_file="${tmake_file} i386/t-x86_64-dgos"
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h newlib-stdint.h i386/x86-64.h dgos.h"
+	;;
 i[34567]86-pc-msdosdjgpp*)
 	xm_file=i386/xm-djgpp.h
 	tm_file="dbxcoff.h ${tm_file} i386/unix.h i386/bsd.h i386/gas.h i386/djgpp.h i386/djgpp-stdint.h"
diff -durN -x Makefile.in -x autom4te.cache -x configure -x crossconfig.m4 orig-src/gcc-10.2.0/libgcc/config.host toolchain_build/src/gcc-10.2.0/libgcc/config.host
--- orig-gcc-10.2.0/libgcc/config.host      2020-09-24 23:03:43.975547463 -0400
+++ gcc-10.2.0/libgcc/config.host   2020-10-07 07:06:36.614530709 -0400
@@ -248,6 +248,14 @@
   tmake_file="$tmake_file t-crtstuff-pic t-libgcc-pic t-eh-dw2-dip t-slibgcc t-slibgcc-fuchsia"
   extra_parts="crtbegin.o crtend.o"
   ;;
+i[34567]86-*-dgos*)
+  extra_parts="crtbegin.o crtend.o"
+  tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic t-eh-dw2-dip i386/t-x86_64-dgos"
+  ;;
+x86_64-*-dgos*)
+  extra_parts="crtbegin.o crtend.o"
+  tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic t-eh-dw2-dip i386/t-x86_64-dgos"
+  ;;
 *-*-linux* | frv-*-*linux* | *-*-kfreebsd*-gnu | *-*-gnu* | *-*-kopensolaris*-gnu | *-*-uclinuxfdpiceabi)
   tmake_file="$tmake_file t-crtstuff-pic t-libgcc-pic t-eh-dw2-dip t-slibgcc t-slibgcc-gld t-slibgcc-elf-ver t-linux"
   extra_parts="crtbegin.o crtbeginS.o crtbeginT.o crtend.o crtendS.o"
@@ -708,6 +716,12 @@
 	    ;;
 	esac
 	;;
+i[34567]86-*-dgos*)
+	tmake_file="$tmake_file t-libgcc-pic"
+	;;
+x86_64-*-dgos*)
+	tmake_file="$tmake_file t-libgcc-pic"
+	;;
 x86_64-*-fuchsia*)
 	tmake_file="$tmake_file t-libgcc-pic"
 	;;
