diff -durN gdb-8.1/gdb/remote-original.c gdb-8.1/gdb/remote.c
--- gdb-8.1/gdb/remote-original.c	2018-06-17 03:03:53.086269371 -0400
+++ gdb-8.1/gdb/remote.c	2018-06-17 03:15:01.954346521 -0400
@@ -7569,7 +7569,7 @@
   buf_len = strlen (rs->buf);

   /* Further sanity checks, with knowledge of the architecture.  */
-  if (buf_len > 2 * rsa->sizeof_g_packet)
+  if (0 && (buf_len > 2 * rsa->sizeof_g_packet))
     error (_("Remote 'g' packet reply is too long (expected %ld bytes, got %d "
 	     "bytes): %s"), rsa->sizeof_g_packet, buf_len / 2, rs->buf);

--- gdb-9.1-orig/gdb/i386-tdep.c	2020-05-30 23:35:32.204688927 -0400
+++ gdb-9.1/gdb/i386-tdep.c	2020-05-30 23:38:10.416447646 -0400
@@ -8514,7 +8514,7 @@
   set_gdbarch_get_longjmp_target (gdbarch, i386_get_longjmp_target);

   /* Call dummy code.  */
-  set_gdbarch_call_dummy_location (gdbarch, ON_STACK);
+  set_gdbarch_call_dummy_location (gdbarch, AT_ENTRY_POINT);
   set_gdbarch_push_dummy_code (gdbarch, i386_push_dummy_code);
   set_gdbarch_push_dummy_call (gdbarch, i386_push_dummy_call);
   set_gdbarch_frame_align (gdbarch, i386_frame_align);
